pkg/workloads/docker.go

```diff
@@ -691,3 +622,81 @@ func (d *dockerClient) GetAllInfraContainersPID() (map[string]int, error) {
 
 	return pids, nil
 }
+
+// finishIpvlanInit finishes configuring ipvlan slave device of the given endpoint.
+//
+// Unfortunately, Docker libnetwork itself moves a netdev to netns of a container
+// (after we respond to a `JoinEndpoint` request in the plugin driver), so
+// qdisc's of the netdev get flushed, thus this complexity below.
+//
+// To make this part idempotent (we have to, because `handleCreateWorkload`
+// can be called many times for the same container in parallel), we can
+// check whether the datapath map has been pinned which indicates previous
+// successful invocation of the function for the same container.
```

Comment:
Nit: could you also add a FIXME describing the race that still needs to be resolved for libnetwork